src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,41 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
+import { UserDTO } from './user.dto';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
   constructor(private userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param('id') id) {
+    const errResp: IResult<UserDTO> = {
+      status: 'NOT OK',
+      error: '',
+      result: null,
+    };
+
+    try {
+      const user = await this.userService.getUserById(id);
+      if (user) {
+        const userResp: UserDTO = {
+          name: user.name,
+          _id: user.id,
+          emailAddress: user.emailAddress,
+          joinedAt: user.joinedAt,
+          phoneNumber: user.phoneNumber,
+        };
+        const resp: IResult<UserDTO> = {
+          status: 'OK',
+          result: userResp,
+        };
+        return resp;
+      }
+      errResp.error = 'User not found';
+      return errResp;
```

Comment:
Declaring the errResp here instead of declaring it as soon as the method is called is better right?

Reply:
Yeah. But that way we'd be declaring two diff error responses, one in try and one in catch..i thought we could do this instead to reduce redundant code

Reply:
But it doesn't make a lot of sense, declaring the error variable right as the first line of method implementation